gcm/src/main/java/com/parse/gcm/ParseGCM.java

```diff
@@ -25,39 +24,25 @@
  */
 public class ParseGCM {
 
-    private static final String SENDER_ID_EXTRA = "com.parse.push.gcm_sender_id";
-
     static final String TAG = "ParseGCM";
 
-    private static final String JOB_TAG_REGISTER = "register";
+    private static final String SENDER_ID_EXTRA = "com.parse.push.gcm_sender_id";
 
     /**
      * Register your app to start receiving GCM pushes
      *
-     * @param context context
+     * @param context     context
      */
-    public static void register(Context context) {
+    public static void register(@NonNull Context context) {
         //kicks off the background job
         PLog.v(TAG, "Scheduling job to register Parse GCM");
         FirebaseJobDispatcher dispatcher = new FirebaseJobDispatcher(new GooglePlayDriver(context.getApplicationContext()));
-        Job job = dispatcher.newJobBuilder()
-                .setRecurring(false)
-                .setReplaceCurrent(true)
-                // retry with exponential backoff
-                .setRetryStrategy(RetryStrategy.DEFAULT_EXPONENTIAL)
-                .setConstraints(
-                        // only run on a network
-                        Constraint.ON_ANY_NETWORK
-                )
-                .setService(ParseGCMJobService.class) // the JobService that will be called
-                .setTag(JOB_TAG_REGISTER)        // uniquely identifies the job
-                .build();
-
+        Job job = ParseGCMJobService.createJob(dispatcher, gcmSenderFromManifest(context));
```

Comment:
You're getting read of manifest but keeping it here?

Reply:
Yeah, I tried to get rid of it for the GCM module, but it turns out we need the GCM sender ID at unpredictable times, such as when the token gets refreshed. So, we would either have to keep it in the manifest, or store it in SharedPreferences or something. I think it being in the manifest as it was before is the easiest configuration. Since most should be migrating away from the GCM module anyways, I think it is okay.